README.md

```diff
@@ -2,31 +2,303 @@
+method to classes that are super classes of the class we specify.
+
+Let's rewrite our `sumList()` method to only accept lists of `Integer` objects:
+
+```java
+public static Integer sumIntegerList(List<Integer> numberList) {
+    int sum = 0;
+    for (Integer number: numberList) {
+        sum = sum + number;
+    }
+
+    return sum;
+}
+```
+
+This would work to add a list of `Integer` objects:
+
+```java
+public static void main(String[] args) {
+    List<Integer> integers = new ArrayList<Integer>();
+    integers.add(10);
+    integers.add(20);
+    integers.add(30);
+
+    Number sum = sumLIntegerist(integers);
+    System.out.println("sum = " + sum);
+}
+```
+
+But what if we wanted our `sumIntegerList()` method to be able to add integers
+that are held in any class that is a superclass of the `Integer` class?
+
+Let's first examine the `Integer` class hierarchy:
+
+- `Integer extends Number` and
+- `Number extends Object`, which yields the following hierarchy:
+- `Object` -> `Number` -> `Integer`
+
+Trying to call this `sumIntegerList()` method with a list of `Number` objects
+like this:
+
+```java
+public static void main(String[] args) {
+    List<Number> numbers = new ArrayList<Number>();
+    numbers.add(0);
+    numbers.add(10);
+    numbers.add(20);
+
+    Number sum = sumIntegerList(numbers);
+    System.out.println("sum = " + sum);
+}
+```
+
+Gives us the following error:
+
+```plaintext
+Example.java:21: error: incompatible types: List<Number> cannot be converted to List<Integer>
+        Integer sum = sumList(nums);
+```
+
+This is because the type `List<Integer>` only matches the `Integer` class
+specifically. For more flexibility, we can specify a lower bound for our generic
+type like so:
+
+```java
+public static Integer sumIntegerList(List<? super Integer> numberList) {     
+    Integer sum = Integer.valueOf(0);
+    for (Object number: numberList) {
+        sum += (Integer)number;
+    }
+
+    return sum;
+}
+```
+
+The syntax for specifying a lower bound wildcard is as follows:
+
+- In the generic parameter type definition, use the `? super Type` notation
+  where `?` is the wildcard and `Type` is the class that should be the lower
+  bound of the parameter type.
+
+In this case, all objects that are passed into this method must be a `Integer`
+or one of the classes it inherits from (directly, i.e. `Number` or indirectly,
+which in this case means `Object`).
+
+Also notice that we had to change the `for` loop slightly too to adjust for a
+lower bound. If, in the event, we receive a Java `Object`, then we need to iterate
+through the list of the Java `Objects`, and then explicitly cast it to an
+`Integer`.
+
+Now the following code compiles and runs:
+
+```java
+import java.util.List;
+import java.util.ArrayList;
+
+public class Example {
+
+  public static Integer sumList(List<? super Integer> numberList) {
+    int sum = 0;
+    for (Object number : numberList) {
+      sum = sum + (Integer)number;
```

Comment:
The BlackRock students were confused by lower-bound generics. They didn't understand why we would want a method that would cast any Object to an Integer, since it seemed to get rid of all type safety.

Would it be possible to change the lower-bound example to something more likely to be seen in a job?